lib/krb5/cache.c

```diff
@@ -768,13 +788,49 @@ krb5_cc_default_name(krb5_context context)
     return context->default_cc_name;
 }
 
+/*
+ * Use this instead of calling `(*ops->get_default_name)(...)' directly.
+ *
+ * Plugins exist (lib/krb5/pcache.c) that don't return the default cache name
+ * with the cctype prefix.
+ */
+static krb5_error_code
+wrap_get_default_name(krb5_context context, const krb5_cc_ops *ops, char **n)
+{
+    krb5_error_code ret;
+    char *s = NULL;
+
+    *n = NULL;
+    ret = (*ops->get_default_name)(context, &s);
+    if (ret == 0) {
+        size_t plen = strlen(ops->prefix);
+
+        if (strncmp(s, ops->prefix, plen) == 0 && s[plen] == ':') {
+            *n = s;
+            s = NULL;
+        } else if (asprintf(n, "%s:%s", ops->prefix, s) == -1 || *n == NULL) {
```

Comment:
If asprintf() returns -1 then `*n` is left in an undefined state.   It might be NULL but it might also be pointing at unallocated memory.   `*n` must be assigned to NULL to be safe.